File: java/jimmer-sql/repository/src/main/java/org/babyfish/jimmer/sql/example/repository/BookStoreRepository.java
package org.babyfish.jimmer.sql.example.repository;

import org.babyfish.jimmer.spring.repo.support.AbstractJavaRepository;
import org.babyfish.jimmer.spring.repository.JRepository;
import org.babyfish.jimmer.sql.JSqlClient;
import org.babyfish.jimmer.sql.example.model.BookStore;
import org.springframework.stereotype.Component;

@Component
public class BookStoreRepository extends AbstractJavaRepository<BookStore, Long> {
    public BookStoreRepository(JSqlClient sql) {
        super(sql);
    }
}

File: kotlin/jimmer-sql-graphql-kt/repository/src/main/kotlin/org/babyfish/jimmer/sql/example/kt/repository/BookStoreRepository.kt
package org.babyfish.jimmer.sql.example.kt.repository

import org.babyfish.jimmer.spring.repo.support.AbstractKotlinRepository
import org.babyfish.jimmer.sql.example.kt.model.BookStore
import org.babyfish.jimmer.sql.example.kt.model.name
import org.babyfish.jimmer.sql.fetcher.Fetcher
import org.babyfish.jimmer.sql.kt.KSqlClient
import org.babyfish.jimmer.sql.kt.ast.expression.`eq?`
import org.springframework.stereotype.Repository

@Repository
class BookStoreRepository(
    sql: KSqlClient
) : AbstractKotlinRepository<BookStore, Long>(sql) {

    fun findByNameLikeOrderByName(
        name: String?,
        fetcher: Fetcher<BookStore>?
    ): List<BookStore> =
        sql.executeQuery(BookStore::class) {
            where(table.name `eq?` name)
            select(table.fetch(fetcher))
        }
}

